src/main/java/com/n1analytics/paillier/PaillierContext.java

```diff
@@ -397,18 +752,53 @@ public EncryptedNumber add(EncryptedNumber operand1, EncryptedNumber operand2)
     return new EncryptedNumber(this, result, exponent1);
   }
 
+  /**
+   * Adds an {@code EncryptedNumber} and an {@code EncodedNumber}. Encrypts the {@code EncodedNumber}
+   * before adding them together. Throws PaillierContextMismatchException if the {@code PaillierContext}
+   * of either {@code operand1} or {@code operand2} does not match this {@code PaillierContext}.
+   *
+   * @param operand1 an {@code EncryptedNumber}.
+   * @param operand2 an {@code EncodedNumber}.
+   * @return the addition result.
+   * @throws PaillierContextMismatchException if the {@code PaillierContext} of either
+   * {@code operand1} or {@code operand2} does not match this PaillirContext.
+   */
   public EncryptedNumber add(EncryptedNumber operand1, EncodedNumber operand2)
           throws PaillierContextMismatchException {
     return add(operand1, encrypt(operand2));
   }
 
+  /**
+   * Adds an {@code EncodedNumber} and an {@code EncryptedNumber}. Encrypts the {@code EncodedNumber}
+   * before adding them together. Throws PaillierContextMismatchException if the {@code PaillierContext}
+   * of either {@code operand1} or {@code operand2} does not match this {@code PaillierContext}.
```

Comment:
I think this is slightly wrong. The `PaillierContextMismatchException` won't get thrown unless the contexts can't be *made* to match - they don't have to be the same initially though?